web/test/import.test.ts

```diff
@@ -0,0 +1,210 @@
+import { AppWebsocket } from '@holochain/client'
+
+let store: any // too complex of a type to mock
+
+let getAppWs: typeof _getAppWs
+let createProfilesZomeApi: typeof _createProfilesZomeApi
+let createProjectsZomeApi: typeof _createProjectsZomeApi
+let installProjectAppAndImport: typeof _installProjectAppAndImport
+let installProjectApp: typeof _installProjectApp
+let importProjectData: typeof _importProjectData
+let simpleCreateProjectMeta: typeof _simpleCreateProjectMeta
+let baseRootState: typeof mockBaseRootState
+let mockGetState: typeof store.getState
+
+let onStep: Parameters<typeof importProjectsData>[2]
+let mockAppWs: AppWebsocket
+let projectMeta: WireRecord<ProjectMeta>
+let mockMigrationData: string
+let mockCellIdString: string
+
+beforeEach(() => {
+  mockAppWs = {} as typeof mockAppWs
+  getAppWs = jest.fn().mockResolvedValue(mockAppWs)
+
+  createProfilesZomeApi = jest.fn().mockReturnValue({
+    profile: {
+      // turn this jest.fn into a var that you can track and assert on
+      createWhoami: jest.fn(),
+    },
+  })
+  projectMeta = mockUnmigratedProjectMeta
+  createProjectsZomeApi = jest.fn().mockReturnValue({
+    projectMeta: {
+      simpleCreateProjectMeta: jest.fn().mockResolvedValue(projectMeta),
+    },
+  })
+  installProjectAppAndImport = jest.fn()
+  mockCellIdString =
+    '132,45,36,204,129,221,8,19,206,244,229,30,210,95,157,234,241,47,13,85,105,207,55,138,160,87,204,162,244,122,186,195,125,254,5,185,165,224,66[:cell_id_divider:]132,32,36,97,138,27,24,136,8,80,164,189,194,243,82,224,72,205,215,225,2,27,126,146,190,40,102,187,244,75,191,172,155,196,247,226,220,92,1'
+
+  installProjectApp = jest
+    .fn()
+    .mockResolvedValue([mockCellIdString, ['abc'], 'testString'])
+
+  importProjectData = jest.fn()
+  simpleCreateProjectMeta = jest.fn()
+
+  onStep = jest.fn()
+
+  baseRootState = mockBaseRootState
+
+  mockMigrationData = JSON.stringify(sampleGoodDataExport)
+
+  mockGetState = jest.fn().mockReturnValue(baseRootState)
+
+  store = {
+    dispatch: jest.fn(),
+    getState: mockGetState,
+  }
+})
+
+describe('importProjectsData()', () => {
+  it('correctly calls functions with expected arguments', async () => {
+    await internalImportProjectsData(
+      getAppWs,
+      createProfilesZomeApi,
+      installProjectAppAndImport,
+      installProjectApp,
+      store,
+      mockMigrationData,
+      onStep
+    )
+
+    expect(getAppWs).toHaveBeenCalledTimes(1)
+
+    expect(createProfilesZomeApi).toHaveBeenCalledTimes(1)
+    expect(createProfilesZomeApi).toHaveBeenCalledWith({})
+
+    // make sure the test data is actually set up the way the test
+    // expects it to be
+    expect(sampleGoodDataExport.projects.length).toBe(2)
+    expect(sampleGoodDataExport.projects[0].projectMeta.isMigrated).toBeNull()
+    expect(
+      sampleGoodDataExport.projects[1].projectMeta.isMigrated
+    ).not.toBeNull()
+
+    expect(installProjectAppAndImport).toHaveBeenCalledTimes(1)
```

Comment:
Describe these expectations in a comment. How this project needs to be migrated because it wasnt already